conda/environments/env.yml

```diff
@@ -16,8 +16,12 @@ dependencies:
   - scipy
   - python=3.8
   - cudatoolkit=11.0
+  - cudatoolkit-dev=11.0
```

Comment:
Why is this needed?

Reply:
In Conda package, cudatoolkit package doesn't include `nvcc` binary, but nvcc is needed to build the project so added it. Does it make sense? Any other solution to do that? :)

Reply:
Yeah it doesn't. We are working on a longer term solution ( https://github.com/conda-forge/cudatoolkit-feedstock/issues/62 )

However this package was added for historical reasons to help users run the CUDA installer. I don't think we should be using it

Reply:
@jakirkham for now, what would be a good solution until the long-term solution is available?
Currently, without installing cudatoolkit-dev, it would try to find system's nvcc if available which cannot be isolated in the environment. Do you think we can keep the change(`cudatoolkit-dev`) until the long-term solution is available?

Reply:
Is it ok if we just rely on the system one for now? Or are there other issues with that?

Should add that others may just be using our prebuilt packages. So think this is more a question of what works for us

Reply:
> Is it ok if we just rely on the system one for now? Or are there other issues with that?
I thought `env.yml` is for setting up 'development environment' to build cuCIM locally from the source code and it is not for the runtime environment (it already has more than needed for just running cuCIM. e.g., yasm/compilers/cmake/flake8 etc).
 
I don't think all users (developer who want to contribute here) have the same nvcc version needed (in this case, CUDA 11.0) in the system so I think `cudatoolkit-dev` with the same version of `cudatoolkit` is needed.

May be we need to separate out `env.yml` -- one for dev, one for runtime.